masonry/src/widget/text_area.rs

```diff
@@ -299,6 +299,25 @@ impl<const EDITABLE: bool> TextArea<EDITABLE> {
     }
 }
 
+// --- MARK: HELPERS ---
+impl<const EDITABLE: bool> TextArea<EDITABLE> {
+    /// Get the IME area from the editor, accounting for padding.
+    ///
+    /// This should only be called when the editor layout is available.
+    fn ime_area(&self) -> Rect {
```

Comment:
This being repeated so often to me suggests that it should be a method on `Widget`, so that there is only one source of truth needed. That would be a bit of a bigger change to wire up, and I don't have all the specifics.

That is, this is a non-blocking conern